lib/r10k/action/deploy/module.rb

```diff
@@ -103,11 +105,10 @@ def visit_environment(environment)
           else
             logger.debug1(_("Updating modules %{modules} in environment %{env_path}") % {modules: @settings.dig(:overrides, :modules, :requested_modules).inspect, env_path: environment.path})
 
-            environment.deploy
+            updated_modules = environment.deploy
 
-            requested_mods = @settings.dig(:overrides, :modules, :requested_modules) || []
             # We actually synced a module in this env
-            if !((environment.modules.map(&:name) & requested_mods).empty?)
+            if !(updated_modules.nil? || updated_modules.empty?)
```

Comment:
(nit) might be clearer to apply De Morgan's law here:
```suggestion
            if !updated_modules.nil? && !updated_modules.empty?
```